libmscore/measure.cpp

```diff
@@ -261,8 +261,16 @@ Measure::Measure(const Measure& m)
 
 void Measure::layoutStaffLines()
       {
-      for (MStaff* ms : _mstaves)
-            ms->lines()->layout();
+      int staffIdx = 0;
+      for (MStaff* ms : _mstaves) { 
+            if (isCutawayClef(staffIdx) && (score()->staff(staffIdx)->cutaway() || !visible(staffIdx)))
```

Comment:
there's a function in the `Measure` class for that: `Measure::visible(int staffIdx)`

Reply:
Not sure if it's more appropriate though.

Reply:
It's being used there. Means: if the measure has only a clef before end barline, and either (the staff is cutaway or the measure is invisible), then go ahead and draw the staff lines. It will later draw only a small length of the lines to hold the 'courtesy' cutaway clef. 
Otherwise there will be no staff lines at all